### PR TITLE
Increase maximum number of alternative origins to 100

### PR DESCRIPTION
**Pull request by aterga**
# Motivation

The `/.well-known/ii-alternative-origins` file was capped at 10 entries, which is
restrictive for services that legitimately run many frontends under a single
derivation origin. This raises the cap to 100.

# Changes

- `src/frontend/src/lib/utils/validateDerivationOrigin.ts`: `MAX_ALTERNATIVE_ORIGINS`
  raised from `10` to `100`. The user-facing error message interpolates the constant,
  so it now reports the new limit automatically.
- `docs/ii-spec.mdx`: updated the three places stating the limit — the requirements
  list, the JSON schema's `maxItems`, and the "to prevent misuse" note.
- `src/frontend/tests/e2e-playwright/routes/authorize/alternativeOrigins.spec.ts`:
  the "too many entries" test now lists 101 origins. The requesting origin is one
  of them, so the entry limit is the only thing that can reject the sign-in —
  previously the fixture listed only unrelated origins, meaning the test failed at
  the "origin not allowed" check and would have kept passing with the limit removed.
- `src/frontend/src/lib/utils/validateDerivationOrigin.test.ts`: new unit cases for
  both sides of the boundary — exactly 100 entries validates, 101 is rejected with
  the "has too many entries" message. The expected maximum is spelled out in the
  test rather than imported from the implementation, so lowering the limit fails
  the test instead of silently following along.

No backend/Rust change was needed — the limit is enforced only in the frontend
validation of the fetched `ii-alternative-origins` document.

# Tests

- `validateDerivationOrigin.test.ts`: 25/25 pass, including the two new boundary cases.
- Mutation-checked the boundary coverage: temporarily setting the constant back to
  `10` fails the 100-entry case; restoring `100` makes the file green again.
- `prettier --check`, `eslint --max-warnings 0`, and
  `tsc --project tsconfig.all.json --noEmit` are clean on the changed files.
- The Playwright e2e suite was not run locally (it needs a replica and the built
  canisters); CI covers it.
- Unrelated pre-existing local failures in `findWebAuthnFlows`, `iiConnection`, and
  `appMetadata` reproduce identically without these changes — they are environment
  gaps in the sandbox, not regressions.

### CODE REVIEW

File: src/frontend/src/lib/utils/validateDerivationOrigin.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { remapToLegacyDomain } from "$lib/utils/iiConnection";
 import { wrapError } from "$lib/utils/utils";
 import { Principal } from "@icp-sdk/core/principal";
 
-const MAX_ALTERNATIVE_ORIGINS = 10;
+const MAX_ALTERNATIVE_ORIGINS = 100;
```
[the remark this answers is not in the document]
Comment (aterga) [reply src/frontend/src/lib/utils/validateDerivationOrigin.ts:6]: Good catch — fixed in ceead54. Added two unit cases in `validateDerivationOrigin.test.ts`: exactly 100 entries (with the request origin among them) validates, and 101 is rejected with the "has too many entries" message, so the assertion can't pass via the "origin not allowed" branch.

The expected maximum is spelled out in the test rather than imported from the implementation, precisely so a stale cap fails it. Verified by temporarily setting the constant back to 10: the 100-entry case fails, and restoring 100 makes all 25 tests in the file pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/frontend/tests/e2e-playwright/routes/authorize/alternativeOrigins.spec.ts [hunk 2]
```diff
@@ -131,21 +131,12 @@ test("Should not issue delegation when /.well-known/ii-alternative-origins has t
...
-      "https://a6.com",
-      "https://a7.com",
-      "https://a8.com",
-      "https://a9.com",
-      "https://a10.com",
-    ],
+    alternativeOrigins: Array.from(
+      { length: 101 },
+      (_, index) => `https://a${index}.com`,
+    ),
```
[the remark this answers is not in the document]
Comment (aterga) [reply src/frontend/tests/e2e-playwright/routes/authorize/alternativeOrigins.spec.ts]: Correct, and it was true of the original 11-entry fixture too. Fixed in ceead54: the list is now 100 filler origins plus `TEST_APP_URL`, so the requesting origin is present and the entry limit is the only thing left that can reject the sign-in.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by sea-snake — APPROVED** [review]
